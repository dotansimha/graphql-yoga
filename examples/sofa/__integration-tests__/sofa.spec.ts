import { yoga } from '../src/yoga.js'

describe('SOFA', () => {
  it('serve Swagger UI correctly', async () => {
    const response = await yoga.fetch('http://localhost:4000/swagger')
    expect(response.status).toBe(200)
    expect(response.headers.get('Content-Type')).toBe('text/html')
    const html = await response.text()
    expect(html).toContain('<title>SwaggerUI</title>')
  })
  it('serve Swagger JSON correctly', async () => {
    const response = await yoga.fetch('http://localhost:4000/swagger.json')
    expect(response.status).toBe(200)
    expect(response.headers.get('Content-Type')).toBe('application/json')
    const json = await response.json()
    expect(json).toMatchSnapshot('sofa-swagger')
  })
  it('serve GraphiQL correctly', async () => {
    const response = await yoga.fetch('http://localhost:4000/graphql', {
      method: 'GET',
      headers: {
        Accept: 'text/html',
      },
    })
    expect(response.status).toBe(200)
    expect(response.headers.get('Content-Type')).toBe('text/html')
    const html = await response.text()
    expect(html).toContain('<title>Yoga GraphiQL</title>')
  })
  it('serve GraphQL API correctly', async () => {
    const response = await yoga.fetch('http://localhost:4000/graphql', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: `
          query {
            users {
              id
              name
            }
          }
        `,
      }),
    })
    expect(response.status).toBe(200)
    const json = await response.json()
    expect(json).toMatchSnapshot('sofa-graphql')
  })
  it('serve REST API correctly', async () => {
    const response = await yoga.fetch('http://localhost:4000/rest/me')
    expect(response.status).toBe(200)
    expect(response.headers.get('Content-Type')).toBe('application/json')
    const json = await response.json()
    expect(json).toMatchSnapshot('sofa-rest-me')
  })
})
