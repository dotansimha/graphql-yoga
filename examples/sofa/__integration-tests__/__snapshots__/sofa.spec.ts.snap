// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SOFA serve GraphQL API correctly: sofa-graphql 1`] = `
{
  "data": {
    "users": [
      {
        "id": "1",
        "name": "User A",
      },
      {
        "id": "2",
        "name": "User B",
      },
    ],
  },
}
`;

exports[`SOFA serve REST API correctly: sofa-rest-me 1`] = `
{
  "favoriteBook": {
    "id": "1",
  },
  "favoriteFood": {
    "dough": "classic",
    "toppings": [
      "ham",
    ],
  },
  "favoritePizza": {
    "dough": "pan",
    "toppings": [
      "cheese",
    ],
  },
  "id": "1",
  "name": "User A",
  "shelf": [
    {
      "id": "1",
    },
    {
      "id": "2",
    },
  ],
}
`;

exports[`SOFA serve Swagger JSON correctly: sofa-swagger 1`] = `
{
  "components": {
    "schemas": {
      "Book": {
        "properties": {
          "id": {
            "type": "string",
          },
          "title": {
            "type": "string",
          },
          "type": {
            "enum": [
              "AUDIO",
              "LEGACY",
            ],
            "type": "string",
          },
        },
        "required": [
          "id",
          "title",
          "type",
        ],
        "type": "object",
      },
      "Mutation": {
        "properties": {
          "addBook": {
            "$ref": "#/components/schemas/Book",
          },
        },
        "type": "object",
      },
      "Pizza": {
        "properties": {
          "dough": {
            "type": "string",
          },
          "toppings": {
            "items": {
              "type": "string",
            },
            "type": "array",
          },
        },
        "required": [
          "dough",
        ],
        "type": "object",
      },
      "Post": {
        "properties": {
          "comments": {
            "items": {
              "type": "string",
            },
            "type": "array",
          },
        },
        "type": "object",
      },
      "Query": {
        "properties": {
          "book": {
            "$ref": "#/components/schemas/Book",
          },
          "books": {
            "items": {
              "$ref": "#/components/schemas/Book",
            },
            "type": "array",
          },
          "feed": {
            "items": {
              "$ref": "#/components/schemas/Post",
            },
            "type": "array",
          },
          "me": {
            "$ref": "#/components/schemas/User",
            "description": "Resolves current user",
          },
          "never": {
            "type": "string",
          },
          "user": {
            "$ref": "#/components/schemas/User",
          },
          "users": {
            "items": {
              "$ref": "#/components/schemas/User",
            },
            "type": "array",
          },
          "usersLimit": {
            "items": {
              "$ref": "#/components/schemas/User",
            },
            "type": "array",
          },
          "usersSort": {
            "items": {
              "$ref": "#/components/schemas/User",
            },
            "type": "array",
          },
        },
        "type": "object",
      },
      "Salad": {
        "properties": {
          "ingredients": {
            "items": {
              "type": "string",
            },
            "type": "array",
          },
        },
        "required": [
          "ingredients",
        ],
        "type": "object",
      },
      "Subscription": {
        "properties": {
          "onBook": {
            "$ref": "#/components/schemas/Book",
          },
        },
        "type": "object",
      },
      "User": {
        "properties": {
          "favoriteBook": {
            "$ref": "#/components/schemas/Book",
          },
          "favoriteFood": {
            "type": "object",
          },
          "favoritePizza": {
            "$ref": "#/components/schemas/Pizza",
          },
          "id": {
            "type": "string",
          },
          "name": {
            "type": "string",
          },
          "shelf": {
            "items": {
              "$ref": "#/components/schemas/Book",
            },
            "type": "array",
          },
        },
        "required": [
          "id",
          "name",
          "favoritePizza",
          "favoriteBook",
          "favoriteFood",
          "shelf",
        ],
        "type": "object",
      },
    },
  },
  "info": {
    "title": "Example API",
    "version": "3.0.0",
  },
  "openapi": "3.0.0",
  "paths": {
    "/rest/add-book": {
      "post": {
        "description": "",
        "operationId": "addBook_mutation",
        "parameters": [],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "properties": {
                  "title": {
                    "type": "string",
                  },
                },
                "required": [
                  "title",
                ],
                "type": "object",
              },
            },
          },
        },
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Book",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/book/{id}": {
      "get": {
        "description": "",
        "operationId": "book_query",
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Book",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/books": {
      "get": {
        "description": "",
        "operationId": "books_query",
        "parameters": [],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/Book",
                  },
                  "type": "array",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/feed": {
      "get": {
        "description": "",
        "operationId": "feed_query",
        "parameters": [
          {
            "in": "query",
            "name": "feed_comments_filter",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/Post",
                  },
                  "type": "array",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/me": {
      "get": {
        "description": "Resolves current user",
        "operationId": "me_query",
        "parameters": [],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/User",
                },
              },
            },
            "description": "Resolves current user",
          },
        },
        "summary": "Resolves current user",
        "tags": [],
      },
    },
    "/rest/never": {
      "get": {
        "description": "",
        "operationId": "never_query",
        "parameters": [],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "type": "string",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/user/{id}": {
      "get": {
        "description": "",
        "operationId": "user_query",
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/User",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/users": {
      "get": {
        "description": "",
        "operationId": "users_query",
        "parameters": [],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/User",
                  },
                  "type": "array",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/users-limit": {
      "get": {
        "description": "",
        "operationId": "usersLimit_query",
        "parameters": [
          {
            "in": "query",
            "name": "limit",
            "required": true,
            "schema": {
              "format": "int32",
              "type": "integer",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/User",
                  },
                  "type": "array",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
    "/rest/users-sort": {
      "get": {
        "description": "",
        "operationId": "usersSort_query",
        "parameters": [
          {
            "in": "query",
            "name": "sort",
            "required": true,
            "schema": {
              "type": "boolean",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/User",
                  },
                  "type": "array",
                },
              },
            },
            "description": "",
          },
        },
        "summary": "",
        "tags": [],
      },
    },
  },
  "servers": [
    {
      "description": "Development server",
      "url": "/",
    },
  ],
  "tags": [],
}
`;
